sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnarBatchIterator.scala

```diff
@@ -106,12 +106,17 @@ class GpuColumnarBatchWithPartitionValuesIterator(
 
   override def next(): ColumnarBatch = {
     if (!hasNext) throw new NoSuchElementException()
+    val isToAddPartitionCols = partSchema.nonEmpty
```

Comment:
Nit: This name is a bit confusing.  I suggest `needPartitionCols` or simply omit it and use `partSchema.nonEmpty` directly.

Reply:
Define this `isToAddPartitionCols` before getting a batch in case of leaking the batch when `partSchema` is null. I will update the name.